V compiler fails to pass struct method to C

### Describe the bug

V compiler fails to pass struct method to C.

### Reproduction Steps

1. Import a function from C
2. Create an interface
3. Create a struct that implements said interface
4. Create a function that takes generic type but no arguments
5. Use that function to create she Struct and pass its method to a function that is imported from C

Small sample that reproduces the bug: https://github.com/veranovus/v-bug-report

### Expected Behavior

I was expecting struct's method to be passed to the C function.

### Current Behavior

```
================== C compilation error (from cc): ==============
cc: /tmp/v_501/bug_report.01J47TN359BPQMPFN4DM6KCKSM.tmp.c:12864:12: error: use of undeclared identifier 'main__Module_import_'; did you mean 'main__TestModule_import_'?
cc:         test_func(main__Module_import_, ((char*)(_SLIT("TestModule").str)));
cc:                   ^~~~~~~~~~~~~~~~~~~~
cc:                   main__TestModule_import_
cc: /tmp/v_501/bug_report.01J47TN359BPQMPFN4DM6KCKSM.tmp.c:12854:22: note: 'main__TestModule_import_' declared here
cc: VV_LOCAL_SYMBOL void main__TestModule_import_(main__TestModule mod) {
cc:                      ^
cc: 1 error generated.
================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

### Possible Solution

No idea, fix compiler ^^

### Additional Information/Context

_No response_

### V version

`V 0.4.7 a4b8768`

### Environment details (OS name and version, etc.)

```
V full version: V 0.4.7 d43f0e4.a4b8768
OS: macos, macOS, 14.3.1, 23D60
Processor: 8 cpus, 64bit, little endian, Apple M1

getwd: /Users/vera/Documents/V/bug_report
vexe: /Users/vera/Documents/V/v/v
vexe mtime: 2024-08-01 20:39:11

vroot: OK, value: /Users/vera/Documents/V/v
VMODULES: OK, value: /Users/vera/.vmodules
VTMP: OK, value: /tmp/v_501

Git version: git version 2.39.3 (Apple Git-145)
Git vroot status: weekly.2024.31-25-ga4b87681
.git/config present: true

CC version: Apple clang version 15.0.0 (clang-1500.1.0.2.5)
thirdparty/tcc status: thirdparty-macos-arm64 713692d4
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -3981,7 +3981,14 @@ fn (mut g Gen) selector_expr(node ast.SelectorExpr) {
 		}
 		if !has_embeds {
 			if !node.has_hidden_receiver {
-				g.write('${g.typ(node.expr_type.idx())}_${m.name}')
+				if node.expr is ast.Ident && sym.info is ast.Interface {
+					left_cc_type := g.cc_type(g.table.unaliased_type(node.expr_type),
+						false)
+					left_type_name := util.no_dots(left_cc_type)
+					g.write('${c_name(left_type_name)}_name_table[${node.expr.name}${g.dot_or_ptr(node.expr_type)}_typ]._method_${m.name}')
+				} else {
+					g.write('${g.typ(node.expr_type.idx())}_${m.name}')
+				}
 				return
 			}
 			receiver := m.params[0]

--- a/vlib/v/tests/interface_var_selector_test.v
+++ b/vlib/v/tests/interface_var_selector_test.v
@@ -1,0 +1,28 @@
+interface Module {
+	import_()
+}
+
+struct TestModule {}
+
+fn (mod TestModule) import_() {
+	println("I'm called!")
+}
+
+fn t_func(a voidptr, b &char) bool {
+	return true
+}
+
+fn import_module[T]() bool {
+	$if T !is Module {
+		println("Type doesn't implement Module.")
+		return
+	}
+
+	mod := Module(T{})
+
+	return t_func(mod.import_, &char(T.name.str))
+}
+
+fn test_main() {
+	assert import_module[TestModule]() == true
+}

